client/src/components/users/types.ts

```diff
@@ -89,9 +91,23 @@ export class UserCollection {
     });
   }
 
+  getSearchFilterUsers(): Array<UserModel> {
```

Comment:
You don't need to add all of this. `getUsers()` already returns the visible users only. So just add the searchFilter to `userIsVisible`:
```
if (!user.humanHandle.label.toLocaleLowerCase().includes(this.searchFilter) &&
    !user.humanHandle.email.toLocaleLowerCase().includes(this.searchFilter) {
  return false;
}
```
and everything should take care of itself.

Since we want `searchFilter` to also be lower case, you could rename the attribute `_searchFilter` and add a setter that makes it lowerCase:

```
public set searchFilter(value: string) {
  this._searchFilter = value.toLocaleLowerCase();
}
```